### PR TITLE
feat: add support for bzlmod

### PR DESCRIPTION
**Pull request by luispadron**
Adds support for bzlmod

### CODE REVIEW

**Comment by luispadron** [comment]
@mattrobmattrob Pushed up a fix for some things i ran when testing e2e:

- Split `SwiftBEPParser/BUILD` into `SwiftBEPParser/Sources/BUILD` and `SwiftBEPParser/Tests/BUILD`. This removes the need to use `package(default_visibility = ["@rules_xcodeproj//xcodeproj:generated"])` for the library package (which would require declaring `rules_xcodeproj` a non-dev dependency)
- Make `SwiftBEPParser/Sources:SwiftBEPParser` public so i can import it in my tool

File: .bazelrc [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## BUILD
 
+# Enable bzlmod
+common --enable_bzlmod
+
```
Comment (mattrobmattrob) [inline .bazelrc:5]:
```suggestion
## COMMON

# Enable bzlmod
common --enable_bzlmod

## BUILD

```

File: MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,98 @@
+"""Defines the bzlmod module"""
+
+module(
+    name = "swift_bep_parser",
+    version = "0.0.0",
+    bazel_compatibility = [
+        ">=6.0.0",
+        "<7.0.0",
```
Comment (mattrobmattrob) [inline MODULE.bazel:8]: Any thoughts about making this `< 8.0.0`, @luispadron?
```suggestion
        "<8.0.0",
```

**Review by mattrobmattrob — APPROVED** [review]

File: MODULE.bazel (hunk 2 above)
Comment (luispadron) [reply MODULE.bazel:8]: I wasn't sure if BEP was compatible between 6 & 7 but if it is that makes sense!

Comment (mattrobmattrob) [reply MODULE.bazel:8]: We're technically using the BEP definition from Bazel 8 (AFAIK) so let's do it.

**Comment by mattrobmattrob** [comment]
I will handle pushing to BCR soon.

**Pull request merged**